lib/puppet/type/truststore_certificate.rb

```diff
@@ -0,0 +1,46 @@
+Puppet::Type.newtype(:truststore_certificate) do
+  desc 'adds a certificate to a pkcs12 truststore'
+
+  ensurable
+
+  newparam(:name, :namevar => true) do
+    desc "The certificate alias used to store inside the truststore"
+  end
+
+  newparam(:truststore) do
+    desc "Path to the truststore to use or create when importing the certificate"
+    isrequired
+  end
+
+  newproperty(:certificate) do
+    desc "Path to the certificate to add to the truststore"
+
+    def fingerprint(file)
+      provider.fingerprint(file)
+    end
+
+    def should_to_s(newvalue)
+      self.class.format_value_for_display(fingerprint(newvalue))
+    end
+
+    def insync?(is)
+      is == fingerprint(should)
+    end
+  end
+
+  newparam(:password_file) do
+    desc "Path to file containing the truststore password"
+    isrequired
+
+    validate do |value|
+      # TODO: Often provider is not set. Only if prefetch is implemented
```

Comment:
I did have a prefetch implementation in the past, but I'm not entirely sure how the lifecycle would be. If it tries to read it at the beginning of the catalog, it doesn't exist yet and neither do the actual resources.